lib/cabin/rspec/cabin_helper.rb

```diff
@@ -0,0 +1,113 @@
+module CabinHelper
+  class LoggingReceiver
+    def filter(&block)
+      cache.detect(&block)
+    end
+
+    def cache
+      @cache ||= []
+    end
+
+    def <<(hash)
+      cache.push(hash)
+    end
+
+    def reset
+      cache.clear
+    end
+  end
+
+  class ReceiveLogMessage
+    def initialize(receiver, message, query)
+      @receiver = receiver
+      @message = message
+      @query = query
+      @compound = false
+      prepare_query
+    end
+
+    def matches?(actual)
+      if actual.is_a?(LoggingReceiver)
+        actual.filter(&method(:filter_block))
+      elsif actual.is_a?(Proc)
+        actual.call
+        @receiver.filter(&method(:filter_block))
+      else
+        @receiver.filter(&method(:filter_block))
+      end
+    end
+
+    def failure_message
```

Comment:
wondering why having this as methods and not as constants? is there any reason for that? please educate me :+1: